src/comestible_inv.cpp

```diff
@@ -0,0 +1,780 @@
+
+    head = catacurses::newwin( head_height, w_width - minimap_width, point( colstart, headstart ) );
+    mm_border = catacurses::newwin( minimap_height + 2, minimap_width + 2,
+                                    point( colstart + ( w_width - ( minimap_width + 2 ) ), headstart ) );
+    minimap = catacurses::newwin( minimap_height, minimap_width,
+                                  point( colstart + ( w_width - ( minimap_width + 1 ) ), headstart + 1 ) );
+    window = catacurses::newwin( w_height, w_width, point( colstart, headstart + head_height ) );
+
+    uistate.comestible_save.exit_code = exit_none;
+
+    //pane is set by child
+    int itemsPerPage = w_height - 2 - 5; // 2 for the borders, 5 for the header stuff
+    pane->init( itemsPerPage, window, &squares );
+}
+
+void comestible_inventory::set_additional_info( std::vector<legend_data> data )
+{
+    pane->additional_info = data;
+    pane->needs_redraw = true;
+}
+
+// be explicit with the values
+enum aim_entry {
+    ENTRY_START = 0,
+    ENTRY_VEHICLE = 1,
+    ENTRY_MAP = 2,
+    ENTRY_RESET = 3
+};
+
+bool comestible_inventory::show_sort_menu( )
+{
+    uilist sm;
+    sm.text = _( "Sort by... " );
+    pane->add_sort_entries( sm );
+    sm.query();
+
+    if( sm.ret < 0 ) {
+        return false;
+    }
+
+    pane->sortby = static_cast<comestible_inv_columns>( sm.ret );
+    return true;
+}
+input_context comestible_inventory::register_actions()
+{
+    input_context ctxt( "COMESTIBLE_INVENTORY" );
+    ctxt.register_action( "HELP_KEYBINDINGS" );
+    ctxt.register_action( "QUIT" );
+
+    ctxt.register_action( "UP" );
+    ctxt.register_action( "DOWN" );
+    ctxt.register_action( "PAGE_DOWN" );
+    ctxt.register_action( "PAGE_UP" );
+    ctxt.register_action( "CATEGORY_SELECTION" );
+    ctxt.register_action( "SORT" );
+    ctxt.register_action( "FILTER" );
+    ctxt.register_action( "RESET_FILTER" );
+
+    ctxt.register_action( "TOGGLE_VEH" );
+    ctxt.register_action( "EXAMINE" );
+    ctxt.register_action( "TOGGLE_AUTO_PICKUP" );
+    ctxt.register_action( "TOGGLE_FAVORITE" );
+    ctxt.register_action( "ITEMS_DEFAULT" );
+    ctxt.register_action( "SAVE_DEFAULT" );
+
+    ctxt.register_action( "ITEMS_NW" );
+    ctxt.register_action( "ITEMS_N" );
+    ctxt.register_action( "ITEMS_NE" );
+    ctxt.register_action( "ITEMS_W" );
+    ctxt.register_action( "ITEMS_CE" );
+    ctxt.register_action( "ITEMS_E" );
+    ctxt.register_action( "ITEMS_SW" );
+    ctxt.register_action( "ITEMS_S" );
+    ctxt.register_action( "ITEMS_SE" );
+    ctxt.register_action( "ITEMS_INVENTORY" );
+    ctxt.register_action( "ITEMS_WORN" );
+    ctxt.register_action( "ITEMS_AROUND" );
+    ctxt.register_action( "ITEMS_AROUND_I_W" );
+    ctxt.register_action( "ITEMS_DRAGGED_CONTAINER" );
+
+    //todo
+    ctxt.register_action( "CONSUME_FOOD" );
+    return ctxt;
+}
+
+std::string comestible_inventory::process_actions( input_context ctxt )
```

Comment:
Do you really need a copy of the input context? I suspect a reference will do just fine.